contracts/staking/src/execute.rs

```diff
@@ -104,12 +104,48 @@ pub fn execute_liquid_stake(
     env: Env,
     info: MessageInfo,
     amount: Uint128,
+    original_sender: Option<String>,
     expected_mint_amount: Option<Uint128>,
 ) -> ContractResult<Response> {
     let config = CONFIG.load(deps.storage)?;
 
     check_stopped(&config)?;
 
+    // a non ibc address is 43 chars long
+    if original_sender.is_none() && info.sender.as_str().len() != 43 {
+        return Err(ContractError::MissingIbcStakeAddress {});
+    }
+
+    // if sent via IBC the user needs to provide his osmosis address which we validate and use
+    let mint_to_address = if original_sender.is_some() && info.sender.as_str().len() != 43 {
```

Comment:
Seems to me that the validation is unnecessary since line 115 already verifies:
- the `original_sender` is not none
- the length of `sender` is 43.

Reply:
it checks for the combination not individuals